Shared/Services/LoopringGraphQLService.cs

```diff
@@ -818,7 +818,92 @@ public LoopringGraphQLService()
+            var query = @"
+                        query accountNFTSlotsQuery(
+                        $skip: Int
+                        $first: Int
+                        $orderBy: AccountNFTSlot_orderBy
+                        $orderDirection: OrderDirection
+                        $where: AccountNFTSlot_filter
+                      ) {
+                        accountNFTSlots(
+                            skip: $skip
+                            first: $first
+                            orderBy: $orderBy
+                            orderDirection: $orderDirection
+                            where: $where
+                          ) {
+                            id
+                            balance
+                            nft {
+                              id
+                            }
+                        }
+                     }
+                    ";
+
+            var request = new RestRequest();
+            request.AddHeader("Content-Type", "application/json");
+            JObject jObject = JObject.FromObject(new
+            {
+                query = query,
+                variables = new
+                {
+                    skip = 0,
+                    first = 1000,
+                    orderBy = "lastUpdatedAt",
+                    orderDirection = "desc",
+                    where = new
+                    {
+                        account = accountId,
+                        nft_not = ""
+                    },
+                }
+            });
+            request.AddStringBody(jObject.ToString(), ContentType.Json);
+            try
+            {
+                var response = await _client.PostAsync(request, cancellationToken);
+                JObject jresponse = JObject.Parse(response.Content!);
+                int count = jresponse["data"]!["accountNFTSlots"]!.Count();
+
+                
+                // This sets an upper limit of 2000 NFTS instead of 1000
+                if (count == 1000)
+                {
+                    jObject = JObject.FromObject(new
+                    {
+                        query = query,
+                        variables = new
+                        {
+                            skip = 1000,
+                            first = 1000,
+                            orderBy = "lastUpdatedAt",
+                            orderDirection = "desc",
+                            where = new
+                            {
+                                account = accountId,
+                                nft_not = ""
+                            },
+                        }
+                    });
+                    request.AddStringBody(jObject.ToString(), ContentType.Json);
 
+                    response = await _client.PostAsync(request, cancellationToken);
+                    jresponse = JObject.Parse(response.Content!);
+                    count = count + jresponse["data"]!["accountNFTSlots"]!.Count();
+                }
```

Comment:
Nice, will do

Reply:
Added while loop. Tried the logic with smaller numbers and it worked as expected;
Setting limit to 2000.
If the count is less than 1000 after the first run it will exit the while loop. If it is greater than 1000 and less than 2000 it will exit while loop. Loop will go until maximum of 2000. Can set this higher eventually if it becomes a problem, but I would guess this is not an issue for 99,9%

Reply:
While the duplicate code has been removed, I still see potential improvements. Most importantly, the code probably never has been tested for more than 1000 NFTs, because there's no account out there that has as many (yet).

I'm going to push a version which I think brings some improvements. Please have a look and see if we can agree on this.

Reply:
Super :)

Reply:
Looks like a good refactor. Even though it's only we who use the method, we should check if chinkSize is larger than 1000 cause that would give error in the request. Max chunkSize must be 1000.

Reply:
Personally, I'd just let it fail, but I'm contributing to your PR, so here we go :-)

Reply:
I bet it would never fail since we call it ourselves, but it's a nice to have check. :)